### PR TITLE
docs(javascript): vanilla JS contents from wiki

### PR DESCRIPTION
**Pull request by asudoh**
#### Changelog

**New**

- Doc contents moved from https://github.com/carbon-design-system/carbon/wiki/Global-JavaScript

### CODE REVIEW

**Review by joshblack — APPROVED** [review]

File: packages/components/src/globals/js/misc/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Misc functions
+
+# `event-matches.js`
+
+Event delgation - returns the closest ancestor of the event target (or the event
```
Comment (emyarod) [inline packages/components/src/globals/js/misc/README.md]:
```suggestion
Event delegation - returns the closest ancestor of the event target (or the event
```

File: packages/components/src/globals/js/misc/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+Creates the composed base class for components to build off of
+
+# `on.js`
+
+Abstracts out the `addEventListener` handler to automatically get released when
+`handle.release()` is run.
+
+# `resize.js`
+
+Creates an resize handler tied to requestAnimationFrame for a more optimized
```
Comment (emyarod) [inline packages/components/src/globals/js/misc/README.md]:
```suggestion
Creates a resize handler tied to requestAnimationFrame for a more optimized
```

File: packages/components/src/globals/js/mixins/README.md [hunk 3]
```diff
@@ -319,3 +319,40 @@ types to instantiate components upon.
 
 `.createdByLauncher()` method allows you to run an event handler of the event
 that caused instantiating the component.
+
+# Other mixins
+
+## `evented-state.js`
+
+In our components, often time clicking a UI element implies a certain state
```
Comment (emyarod) [inline packages/components/src/globals/js/mixins/README.md]:
```suggestion
In our components, oftentimes clicking a UI element implies a certain state
```

**Review by emyarod — COMMENTED** [review]
looks good to me, just a few small changes

**Review by mattrosno — APPROVED** [review]
Thanks @asudoh!

**Comment by asudoh** [comment]
Thank you for correcting typos @emyarod!

**Comment by mattrosno** [comment]
I deleted the wiki page now that this has been merged. It's in the wiki's Git history, but pasting here as a backup too.

# Global JavaScript

- Mixins
- Misc

## Mixins

In an effort to abstract out repeated code, we've introduced the concept of `mixins`, functions that build classes. These are combined by our `mixin` (misc/mixin.js) function which reduces a set of mixin classes down to one.

In essence, by using the various `mixin` functions, you create a single class with a set of shared base functions. Every component Class in the library uses some set of mixin to jumpstart writing the component.

Ex.

```js
class Accordion extends mixin(createComponent, initComponentBySearch) { ... }
```

**Required mixin:**
### createComponent
createComponent establishes the base lifecycle methods for every component, as well as some base initialization configuration. Notably, it sets the element into the component WeakMap storage, as well as houses the core static method of `create` .

*Public Methods added*
- release

*Static Methods added*
- create

*Constructor constants*
- this.element
- this.children
- this.options
- this.constructor.components

**Initialization Mixins:**

### init components
All components get initialized through one of three methods - by an event, by a triggering element, or by a selector. There are three accompanying files for each type, all three give an `init` class to the component that consumes it.

- init-component-by-event | event
- init-component-by-launcher | triggering element
- init-component-by-search | selector

*Static Methods added*
- init

*Static Property added*
- forLazyInit

**Event mixins:**
### evented-state
In our components, often time clicking a UI element implies a certain state change. Examples are things like closing menus, opening modals, or changing a page. In our vanilla library, we try to emit CustomEvents for these actions to let the consuming developer respond to them with a callback function.

*Public Methods added*
- changeState

***Required** Private Methods*
- _changeState

### evented-show-hide-state.js
This one adds hide/show methods to your component and kicks off the change state - you're expected to add in the element that triggers the action, as well as a callback to do something with the state change.

*Public Methods Added*
- show
- hide

### track-blur.js
Adds a blur handler to your component - expects a handleBlur method to be added into the consuming component

**Required Public Methods added**
- handleBlur

## Misc functions

- event-matches.js

  Event delgation - returns the closest ancestor of the event target (or the event target itself) which matches the selectors given in parameter.

- get-launching-details.js

  Given an event, returns an object that contains the launchingElement and the launchingEvent (if either exists)

- mixin.js

  Creates the composed base class for components to build off of

- on.js

  Abstracts out the `addEventListener` handler to automatically get released when Component.release() is run.

- resize.js

  Creates an resize handler tied to requestAnimationFrame for a more optimized result. Has an `add` method to add callbacks to the resize handler.

**Pull request merged**